### PR TITLE
8292692: Move MethodCounters inline functions out of method.hpp

### PR DESCRIPTION
**Pull request by matias9927**
The inline functions related to MethodCounters in method.hpp can be moved to the inline file to reduce the number of includes. Verified with tier 1-5 tests.

Below is a comparison of the old and new include statistics:
```
Old
----
scanning    836 methodCounters.hpp
   2 found    836 method.hpp

scanning    837 invocationCounter.hpp
   2 found    836 method.hpp
   3 found    836 methodCounters.hpp
   4 found    649 interp_masm_x86.hpp
   5 found      0 interp_masm_aarch64.hpp
   6 found      0 interp_masm_arm.hpp
   7 found      0 interp_masm_ppc.hpp
   8 found      0 interp_masm_riscv.hpp
   9 found      0 interp_masm_s390.hpp
  10 found      0 interp_masm_zero.hpp

scanning    298 method.inline.hpp
   2 found    286 continuationEntry_x86.inline.hpp
   3 found      0 continuationEntry_aarch64.inline.hpp
   4 found      0 continuationEntry_ppc.inline.hpp
   5 found      0 continuationEntry_riscv.inline.hpp
```

```
New
-----
scanning    304 methodCounters.hpp
   2 found    299 method.inline.hpp

scanning    476 invocationCounter.hpp
   2 found    304 methodCounters.hpp
   3 found    257 methodData.hpp
   4 found      0 interp_masm_aarch64.hpp
   5 found      0 interp_masm_ppc.hpp
   6 found      0 interp_masm_riscv.hpp
   7 found      0 interp_masm_s390.hpp
   8 found      0 interp_masm_zero.hpp

scanning    299 method.inline.hpp
   2 found    286 continuationEntry_x86.inline.hpp
   3 found      0 continuationEntry_aarch64.inline.hpp
   4 found      0 continuationEntry_ppc.inline.hpp
   5 found      0 continuationEntry_riscv.inline.hpp
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292692](https://bugs.openjdk.org/browse/JDK-8292692): Move MethodCounters inline functions out of method.hpp (**Enhancement** - P4)

### Reviewers
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to [dcffac79](https://git.openjdk.org/jdk/pull/15094/files/dcffac797642191f1636e5ab0f0921bdb6780f08)
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [dcffac79](https://git.openjdk.org/jdk/pull/15094/files/dcffac797642191f1636e5ab0f0921bdb6780f08)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [dcffac79](https://git.openjdk.org/jdk/pull/15094/files/dcffac797642191f1636e5ab0f0921bdb6780f08)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15094/head:pull/15094` \
`$ git checkout pull/15094`

Update a local copy of the PR: \
`$ git checkout pull/15094` \
`$ git pull https://git.openjdk.org/jdk.git pull/15094/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15094`

View PR using the GUI difftool: \
`$ git pr show -t 15094`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15094.diff">https://git.openjdk.org/jdk/pull/15094.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15094#issuecomment-1709119238)

### CODE REVIEW

**Review by iklam — APPROVED** [review]
LGTM.

File: src/hotspot/share/oops/method.inline.hpp [hunk 1]
```diff
@@ -107,4 +108,91 @@ inline bool Method::is_special_native_intrinsic() const {
...
+}
+
+// Initialization only
+inline void Method::clear_number_of_breakpoints() {
+  MethodCounters* mcs = method_counters();
+  if (mcs != nullptr) {
+    mcs->clear_number_of_breakpoints();
+  }
+}
+#endif
```
Comment (calvinccheung) [inline src/hotspot/share/oops/method.inline.hpp:142]: Suggestion:
   ` #endif // INCLUDE_JVMTI`

**Review by calvinccheung — APPROVED** [review]
Looks good. Just one nit.

**Review by vnkozlov — APPROVED** [review]
Looks good. I submitted tier1 testing which includes different builds (I don't see link to testing in RFE).

**Comment by matias9927** [comment]
Thank you for the reviews @calvinccheung, @iklam, and @vnkozlov! 
/integrate

**Pull request closed without merging**